apps/dashboard/src/lib/utils/functions/theme.ts

```diff
@@ -1,14 +1,109 @@
-export function setTheme(theme?: string) {
+import { lighten } from 'color2k';
+
+export function setTheme(theme: string) {
+  // this condition is to check if it's a hex code coming from the db or it's a specified theme
+  if (theme && !theme.includes('theme-')) {
+    // this adds the "custom-theme" styles to the head tag
+    injectCustomTheme(theme);
+
+    // this adds the "custom-theme" classname to the head tag 
+    setCustomTheme('theme-custom');
+  } else {
+
+    // if the string passed has "theme" in it, it goes on with the existing logic
+    if (!theme && document.body.className.includes('theme-')) {
+          const regex = /theme-[\w]+/gi;
+          document.body.className = document.body.className.replace(regex, theme || '');
+          return;
+        }
+      
+        // In case theme already exists in dom, don't add
+        if (document.body.className.includes(theme || '')) return;
+      
+        localStorage.setItem('theme', theme || '');
+        document.body.className = document.body.className.concat(' ', theme || '');
+  }
+}
+
```

Comment:
You can simplify this code by making this `else if (!theme && document.body.className.includes('theme-'))` and not need the extra nested if statement

Reply:
alright